File: 1786-count-the-number-of-consistent-strings/1786-count-the-number-of-consistent-strings.java
class Solution {
    public int countConsistentStrings(String allowed, String[] words) {
        int cnt = 0;
        Set<Character> allowedSet = new HashSet<>();
        for (char ch : allowed.toCharArray()) {
            allowedSet.add(ch);
        }
        for (String word : words) {
            boolean isConsistent = true;
            for (char ch : word.toCharArray()) {
                if (!allowedSet.contains(ch)) {
                    isConsistent = false;
                    break;
                }
            }
            if (isConsistent)cnt++;
        }
        return cnt;
    }
}

File: 1786-count-the-number-of-consistent-strings/1786-count-the-number-of-consistent-strings.cpp
class Solution {
public:
    int countConsistentStrings(string allowed, vector<string>& words) {
        int count=0;
        for(int i = 0 ; i < words.size() ; i++)
        {
            int flag=1;
            for(int j = 0 ; j < words[i].size() ; j++)
            {
                char ch = words[i][j];
                const char* ch1 = strchr(allowed.c_str(), ch);
                if(ch1 == NULL)
                {
                    flag = 0;
                    break;
                }
            }
            if(flag == 1)
            {
                count++;
            }
        }
        return count;
    }
};